### DISCUSSION TITLE
Reference State

### DISCUSSION THREAD

**Discussion by WillamysC** [oemof/tespy#435] [Q&A] [answered]
Hi, it is possible to set another reference state for the calculation of the enthalpy and entropy? I'm still learning how to use the library but  have been struggling with the divergence of number got from the solver report and Cengel book tables i'm using as guide. I know this are relatives, I just want to see number I am familiar with. Thanks..

**Comment by fwitte** [comment, marked as answer]
Hi,

I do not have access to a notebook right now to have a more detailed look, how custom reference states could be implemented on the TESPy side. 

Instead I suggest you could import CoolProp's PropsSI function (for pure fluids) and iterate over the connections (nw.conns['object']) and subtract the PropsSI result at reference temperature and pressure from the SI value (c.h.val_SI) of enthalpy and entropy of every connection. This should work for a quick fix, on the TESPy side I'll put something like a network attribute reference state on my list, which will be subtracted in the results processing automatically.

Have a nice weekend

**Reply by WillamysC** [reply]
not sure I understood, but sense i'm here, i'd like to ask if there is a built in method to get the entropy of componentes or connections like there is "#.h.val" for enthalpy.

**Reply by fwitte** [reply]
Entropy is the s attribute of connection. c.s.val and c.s.val_SI
